backend/globaleaks/handlers/admin/user.py

```diff
@@ -77,8 +125,9 @@ def db_create_user(session, tid, user_session, request, language):
         db_set_user_password(session, tid, user, generateRandomPassword(16))
 
     session.add(user)
-
     session.flush()
+    if not wizard and tid == 1:
+        generate_analyst_key_pair(session, user, user_session)
```

Comment:
in relation to principles of  Separation of Concerns (SoC) or Encapsulation Bottom layers should not be aware of upper layers. This code should be implemented out of this module.